src/components/TextInput/TextInput.js

```diff
@@ -111,6 +111,7 @@ function PrivateTextInput({
         onBlur={onBlur}
         value={value}
         data-tid={rest['data-tid']}
+        aria-label={name} // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute
```

Comment:
Is this strictly an internal call to input? Perhaps the index.d.ts does NOT need updating? 🤔

Reply:
Yes, it should be an internal call

> I think my initial take was that we'd need to update the index.d.ts but it looks like these aria-label attributes may be just getting passed through in a `...rest` or similar prop. Please confirm and verify it doesn't break typescript consumers before merging otherwise LGTM -- thanks!